Align tutorial installation instructions with readme

rknuus opened the pull request:
The readme distinguishes between Ubuntu versions. Further the list of
debian packages in the readme differs from the one in the tutorial.
Because the readme is more recent (see 82c94a72) it's presumably more up
to date.

Hopefully this change doesn't drop a debian package which is used only
by the tutorial but not by the readme.

<!--- Describe your changes in detail -->
<!--- What problem does this change solve? -->
<!--- If this PR relates to an issue, include Refs #XXX or Fixes #XXX -->

## PR Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [ ] All new features have been tested
- [ ] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

freakboy3742 reviewed (changes requested):
Thanks for the report and PR, but... ugh... this is a right mess (my mess, not yours :-)

There's two other "sources of truth" - the [Beeware tutorial](https://github.com/beeware/beeware/blob/master/docs/tutorial/tutorial-0.rst#install-dependencies), and the Linux `system_packages` list in the Briefcase template - and, wouldn't you know it... they all have different lists of dependencies 🤦 

There shouldn't be any difference in the system requirements between any of these lists - there's no difference between the dependencies for the tutorial, the README dependencies, and the Briefcase packaging dependencies. The Briefcase template is for Ubuntu 18.04, so I'd guess that's the most authoritative list (at least for ubuntu 18.04) because it's the list that is automated... but honestly, actually testing on clean installs is the only validation approach that is really going to work at this point.

@rknuus Are you in a position to do that validation (i.e., are you able to spin up a clean virtual machine to test the tutorial)? If you can't, I can; let me know and I'll get you a canonical list of dependencies.

rknuus commented:
> @rknuus Are you in a position to do that validation (i.e., are you able to spin up a clean virtual machine to test the tutorial)? If you can't, I can; let me know and I'll get you a canonical list of dependencies.

@freakboy3742 As I'm planning to use toga for a project I'll do the tutorial soon-ish. I could certainly do it in a clean environment to verify the dependencies from the Briefcase template. But before I'd like to do some basic work on the back end - otherwise I might forget the insights from the tutorial before I'll get to the GUI work. So if you don't mind to keep this PR open for perhaps 2 weeks (no promises, though) I'd be happy to test the dependencies. I might even attempt to streamline the information in all locations you mentioned. HTH.

freakboy3742 commented:
@rknuus No problems - and if I get a chance to spin up some VMs and audit the list of dependencies, I'll report what I find here.

rknuus commented:
> There's two other "sources of truth" - the [Beeware tutorial](https://github.com/beeware/beeware/blob/master/docs/tutorial/tutorial-0.rst#install-dependencies), and the Linux `system_packages` list in the Briefcase template - and, wouldn't you know it... they all have different lists of dependencies 

Hi @freakboy3742 . Unfortunately I couldn't find the `system_packages` list, because I couldn't find the Briefcase template in first place. Neither in [toga](https://github.com/beeware/toga) nor in [briefcase](https://github.com/beeware/briefcase). A bit embarrassing. So it would be very helpful if you could point me to the list. Many thanks.

I'll start with the tutorial "as is" in the mean time to familiarize myself with it and would reproduce the steps in a clean docker container later on.

freakboy3742 commented:
@rknuus Apologies - I should have left a link. The pyproject.toml is specified in [this template file](https://github.com/beeware/briefcase-template/blob/v0.3/%7B%7B%20cookiecutter.app_name%20%7D%7D/pyproject.toml).

rknuus commented:
> @rknuus Apologies - I should have left a link. The pyproject.toml is specified in [this template file](https://github.com/beeware/briefcase-template/blob/v0.3/%7B%7B%20cookiecutter.app_name%20%7D%7D/pyproject.toml).

@freakboy3742 Many thanks for the link.

As outcome of my verification attempt I published [a new project](https://github.com/rknuus/toga-dependency-verification) to set up a test environment based on Ubuntu 18.04 and 20.04 each. The 18.04 dependencies are based on the Briefcase template as you suggested. I tried to keep the set of pre-installed packages minimal to avoid ignoring a dependency by accident. I decided to use a vanilla Ubuntu docker container and added a minimal LXDE set up based on https://github.com/mviereck/dockerfile-x11docker-lxde/blob/master/Dockerfile on top as GUI environment. To keep the verification setup as close as possible to the tutorials I set up a non-root user named `user` with password "togatest" as well as a `venv`.

It kinda worked. I ran into the following issues:

- In addition to the Briefcase template dependencies I had to add `python3-dev`, otherwise `pip3 install pycairo` would fail.
- To prevent error `invalid command 'bdist_wheel'` I had to install pip package `wheel`.
- Under certain circumstances building the wheel `pygobject` failed complaining `no module named 'cairo'`. It either worked or didn't work consistently. My workaround was to install `pycairo` before `toga`. Sounds like something is cheesy in one of those two projects.
- For reasons I haven't analyzed the `nested_boxes.py` application doesn't find the icons.

If you agree I'd suggest to mention the first two findings in the tutorial and readme, but don't mention the last two. I don't think it makes sense to add a workaround to the tutorial/readme for a hopefully temporary issue of the cairo setup. The same applies to the icons. Let me know if you disagree.

Now I'll call it a day and might try to update this PR by the insights till Sun, 31-Jan-2021.

rknuus commented:
Hi @freakboy3742. As requested I updated the dependencies in the tutorial and in addition in the readme, which worked for all five tutorial examples on Ubuntu 18.04 and Ubuntu 20.04 based on https://github.com/rknuus/toga-dependency-verification.

Note that the webkit package versions of the [briefcase-template](https://github.com/beeware/briefcase-template/blob/v0.3/%7B%7B%20cookiecutter.app_name%20%7D%7D/pyproject.toml) seem to be outdated. I hope you don't mind that I'd rather leave this to you to verify and fix.

P.s. Pardon the force-pushes. I rebased on master and accidentally published a draft commit, which I then removed again.

freakboy3742 reviewed (approved):
Apologies for the delay in reviewing; this looks great. I picked up a couple of inconsistencies, and I've tested this on clean virtual environments.
